GameServerLib/GameObjects/Spell/Sector/SpellSectorPolygon.cs

```diff
@@ -0,0 +1,196 @@
+                        _trueHalfLength = distY;
+                    }
+                }
+
+                // Reassign with true width/halflength.
+                trueVertice = new Vector2(currVertice.X * _trueWidth, currVertice.Y * _trueHalfLength);
+
+                // Save current vertice for next iteration so we can compare distance.
+                lastVertice = trueVertice;
+                // Assign scaled vertice.
+                _trueVertices[i] = trueVertice;
+            }
+
+            if (_trueWidth > _trueHalfLength)
+            {
+                CollisionRadius = _trueWidth;
+            }
+            else
+            {
+                CollisionRadius = _trueHalfLength;
+            }
+
+            Position = GetTargetPosition();
+
+            // The below section is making sure the sector has a correct direction.
+
+            var goingTo = new Vector2(castInfo.TargetPositionEnd.X, castInfo.TargetPositionEnd.Z) - Position;
+            var dirTemp = Vector2.Normalize(goingTo);
+
+            if (float.IsNaN(dirTemp.X) || float.IsNaN(dirTemp.Y) && Parameters.BindObject != null)
+            {
+                if (float.IsNaN(Parameters.BindObject.Direction.X) || float.IsNaN(Parameters.BindObject.Direction.Y))
+                {
+                    dirTemp = new Vector2(1, 0);
+                }
+                else
+                {
+                    dirTemp = new Vector2(Parameters.BindObject.Direction.X, Parameters.BindObject.Direction.Z);
+                }
+
+                var endPos = Position + (dirTemp * SpellOrigin.GetCurrentCastRange());
+                CastInfo.TargetPositionEnd = new Vector3(endPos.X, 0, endPos.Y);
+                var startPos = Position - (dirTemp * SpellOrigin.GetCurrentCastRange());
+                CastInfo.TargetPosition = new Vector3(startPos.X, 0, startPos.Y);
+            }
+            else if (float.IsNaN(dirTemp.X) || float.IsNaN(dirTemp.Y))
+            {
+                dirTemp = new Vector2(1, 0);
+            }
+
+            Direction = new Vector3(dirTemp.X, 0, dirTemp.Y);
+
+            VisionRadius = SpellOrigin.SpellData.MissilePerceptionBubbleRadius;
+
+            Team = CastInfo.Owner.Team;
+        }
+
+        public override void Move(float diff)
+        {
+            // Change direction to the bind object's facing direction.
+            Direction = Parameters.BindObject.Direction;
+
+            base.Move(diff);
+        }
+
+        /// <summary>
+        /// Filter function which checks if the given collider is within the bounds of a hitbox.
+        /// </summary>
+        /// <param name="collider">Object to check.</param>
+        /// <returns>True/False.</returns>
+        protected override bool FilterCollisions(IGameObject collider)
+        {
+            if (_trueVertices.Length <= 0)
+            {
+                return false;
+            }
+
+            var start = new Vector2(CastInfo.TargetPosition.X, CastInfo.TargetPosition.Z);
```

Comment:
I will open an issue to maybe consider using a math lib to check overlap of polygons with circles (there are probably many algos for these calculations).

Reply:
The reason for choosing circle to line intersection is that it can be applied to any polygon. If you find a lib, make sure it can be applied to any shape with given vertices.